internal/network/machine_network_cidr.go

```diff
@@ -417,6 +450,57 @@ func GetInventoryNetworksByFamily(hosts []*models.Host, log logrus.FieldLogger)
 	return ret, nil
 }
 
+type CidrMetric struct {
+	cidr   string
+	metric int32
+}
+
+func GetDefaultRouteNetworkByFamily(hosts []*models.Host, log logrus.FieldLogger) map[AddressFamily]string {
+	ret := make(map[AddressFamily]string)
+	cidrmap := make(map[AddressFamily][]CidrMetric)
+	var err error
+
+	for _, h := range hosts {
+		if h.Inventory != "" {
+			var inventory models.Inventory
+			err = json.Unmarshal([]byte(h.Inventory), &inventory)
+			if err != nil {
+				log.WithError(err).Warnf("Unmarshal inventory %s", h.Inventory)
+				continue
+			}
+			for _, route := range inventory.Routes {
+				if isDefault, err := IsDefaultRoute(route); !isDefault {
+					if err != nil {
+						log.Error(err)
+					}
+					continue
+				}
+
+				for family, cidrs := range findMatchingCidrs(route, inventory.Interfaces) {
+					metrics := make([]CidrMetric, 0)
+					for _, cidr := range cidrs {
+						metrics = append(metrics, CidrMetric{cidr, route.Metric})
+					}
+					cidrmap[family] = metrics
+				}
+			}
+		}
+	}
+
+	//per family choose the cidr with the minimal metric
+	for family, cidrWithRoutes := range cidrmap {
+		cidrs := cidrWithRoutes
+		sort.SliceStable(cidrs, func(i, j int) bool {
+			return cidrs[i].metric < cidrs[j].metric
```

Comment:
Since it this function is used for SNO exclusively how about I will change its signature to receive a  single host? In that case it will be clear that the cidrmap holds only data relevant to a single source.